kube-client/src/api/entry.rs

```diff
@@ -0,0 +1,394 @@
+//! API helpers for get-or-create and get-and-modify patterns
+//!
+//! [`Api::entry`] is the primary entry point for this API.
+
+// Import used in docs
+#[allow(unused_imports)] use std::collections::HashMap;
+use std::fmt::Debug;
+
+use kube_core::{params::PostParams, Resource};
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::{Api, Result};
+
+impl<K: Resource + Clone + DeserializeOwned + Debug> Api<K> {
+    /// Gets a given object's "slot" on the Kubernetes API, designed for "get-or-create" and "get-and-modify" patterns
+    ///
+    /// This is similar to [`HashMap::entry`], but [`Entry`] must be [`Entry::sync`]ed for changes to be persisted.
+    pub async fn entry<'a>(&'a self, name: &'a str) -> Result<Entry<'a, K>> {
```

Comment:
I feel a `no_run` doc example here would be helpful for users!

Reply:
Done.